pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+    def r_value(self, value):
+        self._r_value = value
+
+    @property
+    def l_value(self):
+        """Inductance value in H."""
+        return self._l_value
+
+    @model.setter
+    def l_value(self, value):
+        self._l_value = value
+
+    @property
+    def c_value(self):
+        """Capacitance value in F."""
+        return self._c_value
+
+    @model.setter
+    def c_value(self, value):
+        self._c_value = value
+
+    def add(self):
+        """Add a pin to the list of components in the Project Manager."""
+        self._circuit.modeler.schematic.o_component_manager.AddSolverOnDemandModel(
+            self.name,
+            [
+                "NAME:CosimDefinition",
+                "CosimulatorType:=",
+                7,
+                "CosimDefName:=",
+                "DefaultIBISNetlist",
+                "IsDefinition:=",
+                True,
+                "Connect:=",
+                True,
+                "Data:=",
+                [],
+                "GRef:=",
+                [],
+            ],
+        )
+
+    def insert(self, x, y, angle=0.0):
+        """
+        Insert a pin at a defined location inside the graphical window.
+
+        Parameters
+        ----------
+        x: float
+            X position of the pin.
+        y: float
+            Y position of the pin.
+        angle: float, optional
+            Angle of the pin. The default value is ``"0.0"``.
+        """
+
+        self._circuit.modeler.schematic.create_component(
+            component_library=None,
+            component_name=self.name,
+            location=[x, y],
+            angle=angle,
+        )
+
+
+class Buffer:
+    def __init__(self, ibis_name, short_name, circuit):
+        self._ibis_name = ibis_name
+        self._short_name = short_name
+        self._circuit = circuit
+
+    @property
+    def name(self):
+        """Full name of the buffer including the ibis filename."""
+        return "{}_{}".format(self.short_name, self._ibis_name)
+
+    @property
+    def short_name(self):
+        """Short name of the buffer without the ibis filename included."""
+        return self._short_name
+
+    def add(self):
+        """Add a buffer to the list of components in the Project Manager."""
+        self._circuit.modeler.schematic.o_component_manager.AddSolverOnDemandModel(
+            self.name,
+            [
+                "NAME:CosimDefinition",
+                "CosimulatorType:=",
+                7,
+                "CosimDefName:=",
+                "DefaultIBISNetlist",
+                "IsDefinition:=",
+                True,
+                "Connect:=",
+                True,
+                "Data:=",
+                [],
+                "GRef:=",
+                [],
+            ],
+        )
+
+    def insert(self, x, y, angle=0.0):
+        """
+        Insert a buffer at a defined location inside the graphical window.
+
+        Parameters
+        ----------
+        x: float
```

Comment:
Leave a space after the parameter name and the colon `:`, otherwise sphinx won’t render the parameter correctly.